client/dashboard/sites/overview-domains-card/index.tsx

```diff
@@ -121,27 +121,40 @@ export default function DomainsCard( {
 } ) {
 	const { data: sitePlan } = useQuery( siteCurrentPlanQuery( site.ID ) );
 	const { data: siteDomains } = useQuery( siteDomainsQuery( site.ID ) );
+	const filteredSiteDomains = useMemo( () => {
+		if ( ! siteDomains || ! siteDomains.find( ( domain ) => domain.is_wpcom_staging_domain ) ) {
```

Comment:
Hmm, I find the code is hard to follow (unless there's comment).

I think a more self-explanatory logic is as follows, what do you think 😄 

```ts
if ((siteDomains || []).find( domain => domain.is_wpcom_staging_domain ){
   // If the site has *.wpcomstaging.com domain, exclude *.wordpress.com
   return siteDomains.filter( domain => domain.is_wpcom_staging_domain || !domain.wpcom_domain);
}
return siteDomains;
```